Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Find.swift

```diff
@@ -558,3 +578,11 @@ private extension PathHierarchy.Node {
             || keys.contains(String(component.name))
     }
 }
+
+private func typesMatch(provided: [Substring], actual: [String]?) -> Bool {
+    guard let actual = actual, provided.count == actual.count else { return false }
+    for (providedType, actualType) in zip(provided, actual) where providedType != "_" && providedType != actualType {
```

Comment:
I for one prefer the `allSatisfy` version as it's a bit more explicit, I assume that `allSatisfy` implements the short circuiting-behavior as well.

Reply:
Either works for me.

Reply:
Updated to use `allSatisfy` and rephrase the condition to a positive expression instead of a negative expression in https://github.com/swiftlang/swift-docc/pull/643/commits/316351cb938f747704ce7228ae03e0c07d93dda7